### PR TITLE
DM-40889: Improve developer documentation

### PR DESCRIPTION
**Pull request by rra**
Document the trick of adding a checksum annotation to force restarts of the deployment. Stop telling people to add development versions to the Chart.yaml appVersion; for development tags, just use overrides of image.tag. Tell people they can sync their application in the science-platform app to revert development changes. Clarify documentation about Chart.yaml and pull secrets.

### CODE REVIEW

File: docs/developers/deploy-from-a-branch.rst [hunk 1]
```diff
@@ -100,6 +100,7 @@ Make your changes, then click :guilabel:`Save`.
 Normally, these changes will immediately take effect.
 Sometimes if you change a ``ConfigMap`` you will need to restart the relevant deployments to pick up that change.
 For instructions on how to do that, see :ref:`branch-deploy-restart`.
+(However, see :ref:`dev-deployment-restart` for a better way.)
```
Comment (jonathansick) [inline docs/developers/deploy-from-a-branch.rst:103]:
```suggestion
(However, see :ref:`dev-deployment-restart` for a better way.)
To configure the chart to make these restarts happen automatically, see :ref:`dev-deployment-restart`.
```

File: docs/developers/deploy-from-a-branch.rst [hunk 2]
```diff
@@ -183,6 +184,9 @@ Once your branch is merged, remember to reset your application's Argo CD ``Appli
 
 #. In the application's page in Argo CD, click on the :guilabel:`Sync` button to redeploy the application from the default branch.
 
+You can also find your application in the ``science-platform`` Argo CD application and sync it from there to reset the default branch and any other settings you changed.
+Either approach works; use whichever you find more convenient.
```
Comment (jonathansick) [inline docs/developers/deploy-from-a-branch.rst:188]:
```suggestion
Alternatively, you can find the application in the ``science-platform`` Argo CD application and sync it from there to reset the default branch and any other settings you changed.
```

File: docs/developers/write-a-helm-chart.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@ Write a Helm chart for an application
 Argo CD manages applications in the Rubin Science Platform through a set of Helm charts.
 Which Helm charts to deploy in a given environment is controlled by the :file:`values.yaml` and :file:`values-{environment}.yaml` files in `/environments <https://github.com/lsst-sqre/phalanx/tree/main/environments/>`__.
```
Comment (jonathansick) [inline docs/developers/write-a-helm-chart.rst:6]: If the intent is to standardize on not using `/` path prefixes, there may be other instances:

```suggestion
Which Helm charts to deploy in a given environment is controlled by the :file:`values.yaml` and :file:`values-{environment}.yaml` files in `environments <https://github.com/lsst-sqre/phalanx/tree/main/environments/>`__.
```

**Review by jonathansick — APPROVED** [review]

File: docs/developers/deploy-from-a-branch.rst (hunk 1 above)
Comment (rra) [reply docs/developers/deploy-from-a-branch.rst:103]: Yes, this is better, but when applying it, I realized that my whole original change was wrong. The checksum is only computed during Helm operations, and this section is on making changes directly in Argo CD, so the restart always has to be done manually. I'll revise it a different way to make that clear.

**Pull request merged**